artifactory/auth/authutils.go

```diff
@@ -0,0 +1,53 @@
+package auth
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"strings"
+)
+
+func ExtractUsernameFromAccessToken(token string) (string, error) {
+	// Separate token parts.
+	tokenParts := strings.Split(token, ".")
+
+	// Decode the payload.
+	if len(tokenParts) != 3 {
+		return "", errorutils.CheckError(errors.New("Received invalid access-token."))
+	}
+	payload, _ := base64.RawStdEncoding.DecodeString(tokenParts[1])
```

Comment:
This method returns an error which should be probably checked.

Reply:
Added an error check.